eng/targets/ResolveReferences.targets

```diff
@@ -110,9 +110,9 @@
     In addition, enforce use of Reference items for projects reference providers.
   -->
   <Target Name="_CheckForReferenceBoundaries" BeforeTargets="CollectPackageReferences;ResolveReferences">
-    <Error
```

Comment:
A bunch of random noise because I was figuring out how to get STJ source generator work. The relevant bits are in the JSInvokable type and DotNetDispatcher. Also includes a edit to StandaloneApp to verify this works.